lib/sacastats_web/templates/character/template.html.heex

```diff
@@ -32,6 +36,59 @@
         <%= if not is_nil(@character_info.outfit) && @character_info.outfit.leader_character_id == @character_info.character_id do %>
           <p data-toggle="tooltip" title={@character_info.outfit.name <> "'s outfit leader"} class="d-block d-md-inline align-top text-center text-md-left">&#11088;</p>
         <% end %>
+
+        <%= cond do %>
+          <% Map.has_key?(@conn.assigns, :user) && is_nil(@user) -> %>
+            <div class="d-none d-md-inline-block pl-0 pl-lg-5">
+              <button class={"btn btn-primary ml-3 character-favorite-btn px-3 py-3"} data-toggle="dropdown" aria-expanded="false" type="button">
+                <%= HtmlEntities.decode("&#9825;	Favorite") %>
+              </button>
+              <div class="dropdown-menu px-2 not-logged-in-dropdown">
+                <h3>Looks like you are not logged in...</h3>
+                <p><%= "Log in to add '#{@character_info.name_first}' to your favorite's list."%></p>
+                <hr/>
+                <a href="/login">Log in</a>
+              </div>
+            </div>
+            <div class="d-inline-block d-md-none w-100">
+              <div class="text-center">
+                <button class={"btn btn-primary ml-3 character-favorite-btn px-3 py-3"} data-toggle="dropdown" aria-expanded="false" type="button">
+                  <%= HtmlEntities.decode("&#9825;	Favorite") %>
+                </button>
+                <div class="dropdown-menu px-2 not-logged-in-dropdown">
+                  <h3>Looks like you are not logged in...</h3>
+                  <p><%= "Log in to add '#{@character_info.name_first}' to your favorite's list."%></p>
+                  <hr/>
+                  <a href="/login">Log in</a>
+                </div>
+              </div>
+            </div>
+          <% !Map.has_key?(@conn.assigns, :user) -> %>
```

Comment:
it doesn't look like this 2nd conditional has a body, can it be removed? This could probably be an if else too if so

Reply:
It could be interchanged with `Map.has_key?(@conn.assigns, :user) && is_nil(@user)`'s nil check as if we don't have the user we don't want to show the button at all because there would not be a way to access it. (Not signed in means `:user == nil`, but on live pages like the session, we don't have access to `@user` for some reason.

Reply:
We do have access to @user, you just have to get it from the session and add it to the assigns (though there's a plug that should be doing that automatically - might have to make sure it's working on liveviews if you get an error trying to access @user in liveviews.
Either way, I think the 2nd conditional doesn't actually do anything since the `true ->` is a catch-all

Reply:
I guess the problem here is we reference `@user` across this template, but in live views `@user` is not defined as `:user` does not exist on the `@conn.assigns` map. Which means the second conditional is being met.

Reply:
Maybe I misunderstood - is there a difference between @user == nil and @user not being on the assigns at all? Don't both mean the user is logged out?

Reply:
Yes, there is a difference. When a user is not logged in on a regular (non-live) view, `@user == nil`. The only time `@user` is not in the assigns was on the live-views and it didn't matter if they were logged in or not.

Reply:
right I think you just need to assign the user from the session on mount - I did this for the outfit polls in `manage.ex` on L47 `|> assign(:user, session["user"] || session[:user])`

Reply:
Ah, okay.